Assets/HoloToolkit/Utilities/Scripts/SceneList.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+#if UNITY_EDITOR
+using UnityEditor.Callbacks;
+#endif
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Provides a list of names of scenes that were included in the Unity build settings.
+    /// This is necessary because SceneManager doesn't provide the names of scenes before they are loaded
+    /// and EditorBuildSettingsScene is only availabe in the editor.
+    /// </summary>
+    public class SceneList : Singleton<SceneList>
+    {
+#pragma warning disable 0414, 0649 // sceneNamesUpdatedByBuild isn't assigned to, but it is set via Unity serialization
+        [SerializeField]
+        [HideInInspector]
+        private List<string> sceneNamesUpdatedByBuild;
+#pragma warning restore 0414, 0649
```

Comment:
Good idea, but I still get "Assets/HoloToolkit/Utilities/Scripts/SceneList.cs(23,30): warning CS0414: The private field `HoloToolkit.Unity.SceneList.sceneNamesUpdatedByBuild' is assigned but its value is never used". In the editor, this list is assigned to, but we never read from it. The only purpose of writing to it is so that Unity will serialize the value that is then used at runtime.